docs/source/llms/tracing/tracing-schema.rst

```diff
@@ -331,3 +331,116 @@ documents from a vector store). The ``RETRIEVER`` span type has the following sc
+        and will also be used in MLflow evaluation in the future. 
+        
+        The type must be ``List[`` :py:class:`RequestMessage <mlflow.types.chat.RequestMessage>` ``]``
+      - This attribute can be conveniently set using the :py:func:`mlflow.tracing.set_span_chat_messages` function. This function
+        will throw a validation error if the data does not conform to the spec.
+    
+    * - **mlflow.chat.tools**
+      - This attribute represents the tools that were available for the chat model to call. In the OpenAI
+        context, this would be equivalent to the `tools <https://platform.openai.com/docs/api-reference/chat/create#chat-create-tools>`_ 
+        param in the Chat Completions API.
+
+        The type must be a list of valid tools. At the moment, the available tool types are:
+
+        * :py:class:`FunctionTool <mlflow.types.chat.FunctionTool>`
+        * :py:class:`UnityCatalogFunctionTool <mlflow.types.chat.UnityCatalogFunctionTool>`
+
+      - This attribute can be conveniently set using the :py:func:`mlflow.tracing.set_span_chat_tools` function. This function
+        will throw a validation error if the data does not conform to the spec.
+
+Please refer to the example below for a quick demonstration of how to use the utility functions described above, as well as
+how to retrieve them using the :py:class:`span.get_attribute() <mlflow.entities.Span.get_attribute>` function:
+
+.. code-block:: python
+
+  import mlflow
+  from mlflow.entities.span import SpanType
+  from mlflow.tracing.constant import SpanAttributeKey
+  from mlflow.tracing import set_span_chat_messages, set_span_chat_tools
+
+  # example messages and tools
+  messages = [
+      {
+          "role": "system",
+          "content": "please use the provided tool to answer the user's questions",
+      },
+      {"role": "user", "content": "what is 1 + 1?"},
+  ]
+
+  tools = [
+      {
+          "type": "function",
+          "function": {
+              "name": "add",
+              "description": "Add two numbers",
+              "parameters": {
+                  "type": "object",
+                  "properties": {
+                      "a": {"type": "number"},
+                      "b": {"type": "number"},
+                  },
+                  "required": ["a", "b"],
+              },
+          },
+      }
+  ]
+
+
+  @mlflow.trace(span_type=SpanType.CHAT_MODEL)
+  def call_chat_model(messages, tools):
+      # mocking a response
+      response = {
+          "role": "assistant",
+          "tool_calls": [
+              {
+                  "id": "123",
+                  "function": {"arguments": '{"a": 1,"b": 2}', "name": "add"},
+                  "type": "function",
+              }
+          ],
+      }
+
+      combined_messages = messages + [response]
+
+      span = mlflow.get_current_active_span()
+      set_span_chat_messages(span, combined_messages)
+      set_span_chat_tools(span, tools)
```

Comment:
```suggestion
      span = mlflow.get_current_active_span()
      span.set_chat_messages(span, combined_messages)
      span.set_chat_tools(span, tools)
```

Is there a reason we don't these functions as an instance method of `Span`?

Reply:
see https://github.com/mlflow/mlflow/pull/14087#discussion_r1887867532 for context, we don't want to crowd the root Span object with schema-specific functions